### PR TITLE
Fix missing root across safe-point

### PR DESCRIPTION
**Pull request by Keno**
As @yuyichao points out, JL_LOCK is a safe point.

### CODE REVIEW

File: src/codegen.cpp [hunk 1]
```diff
@@ -840,11 +841,13 @@ jl_llvm_functions_t jl_compile_linfo(jl_method_instance_t **pli, jl_code_info_t
     //         Get the codegen lock,
     //         And get the source
     if (li->def == NULL) {
+        JL_GC_PUSH1(&src);
```
Comment (yuyichao) [inline src/codegen.cpp]: We are using a local array for the buffer so this has to be in the same or parent scope as the pop.

**Review by yuyichao — CHANGES_REQUESTED** [review]

**Comment by Keno** [comment]
Updated. Probably best viewed with whitespace ignored.

**Review by yuyichao — APPROVED** [review]
Got identical result following the same reorganization.

**Comment by yuyichao** [comment]
Test failure unrelated....

**Pull request merged**